bin/stringfuzzx

```diff
@@ -89,31 +91,88 @@ def main():
         help    = 'seed the random number generator with the current time (default: {})'.format(DEFAULT_RANDOM)
     )
 
+    # get subparsers
+    subparsers = global_parser.add_subparsers(dest='transformer', help='transformer choice')
+    subparsers.required = True
+
+    # fuzz transformer
+    fuzz_parser = subparsers.add_parser(FUZZ, help='fuzz transformer')
+
+    # graft transformer
+    graft_parser = subparsers.add_parser(GRAFT, help='graft transformer')
+
+    # multiply transformer
+    multiply_parser = subparsers.add_parser(MULTIPLY, help='multiply transformer')
+    multiply_parser.add_argument(
+        '--factor',
+        dest    = 'factor',
+        metavar = 'N',
+        type    = int,
+        default = DEFAULT_FACTOR,
+        help    = 'number to multiply literals by (default: {})'.format(DEFAULT_FACTOR)
+    )
+
+    # nop transformer
+    nop_parser = subparsers.add_parser(NOP, help='nop transformer')
+
+    # reverse transformer
+    reverse_parser = subparsers.add_parser(REVERSE, help='reverse transformer')
+
+    # rotate transformer
+    rotate_parser = subparsers.add_parser(ROTATE, help='rotate transformer')
+
+    # translate transformer
+    translate_parser = subparsers.add_parser(TRANSLATE, help='translate transformer')
+    translate_parser.add_argument(
+        '--integers',
+        dest    = 'integer_flag',
+        action  = 'store_true',
+        default = DEFAULT_INTEGER_FLAG,
+        help    = 'Include integers in translation (default: {})'.format(DEFAULT_INTEGER_FLAG)
+    )
+    
+    # unprintable transformer
+    unprintable_parser = subparsers.add_parser(UNPRINTABLE, help='unprintable transformer')
+
     # parse args
     args = global_parser.parse_args()
 
+    # get the transformer function based on args
+    transformer_name = args.transformer
+    transformer      = TRANSFORMERS[transformer_name]
+
+    # get some flags that will get popped from args before they're used
+    input_file      = args.input_file
+    input_language  = args.input_language
+    output_language = args.output_language
+
+    # read input
+    raw_in = args.input_file.read()
+    ast    = parse(raw_in, input_language)
+
     # seed the RNG
     if args.random is True:
         random.seed()
     else:
         random.seed(args.seed)
 
-    # get transformer
-    transformer = TRANSFORMERS[args.transformer]
+    # get args as a dict
+    transformer_args = vars(args)
 
-    # read input
-    raw_in = args.file.read()
-
-    # run transformer
-    try:
-        transformed = transformer(raw_in, args.input_language)
-    except IndexError as e:
```

Comment:
Please restore the error check and `print` to stderr, possibly around the whole block including the calls to `parse`, `transformer`, and `generate`. Seeing a stack trace makes it look like the program crashed, where it might have been a handled error condition.